ipetcare.spa/src/features/userSearch/containers/userSearchPage.tsx

```diff
@@ -73,80 +90,145 @@ const renderOwners = (results: GetSearchResponseDto) => {
   }
 }
 
 export const UserSearchPage = () => {
   const styles = useStyles()
   const dispatch = useDispatch()
   const searchState = useSelector((state: RootState) => state.search)
-  const [query, setQuery] = useState('')
-  const [sortBy, setSortBy] = useState('' as SortBy)
   const [timeout, setTimeoutState] = useState(0 as any)
+  const [who, setWho] = useState(VET)
+  const [request, setRequest] = useState({
+    query: '',
+    sortBy: undefined,
+    page: 1,
+    pageSize: 10,
+  } as GetSearchDto)
 
-  const onInputChange = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
-  ) => {
-    setQuery(e.target.value)
-    const request = {
-      query,
-      getVetsSortBy: sortBy === '' ? undefined : sortBy,
-    } as GetSearchDto
+  useEffect(() => {
+    if (request.query.length > 0) {
+      if (timeout) clearTimeout(timeout)
+      setTimeoutState(
+        setTimeout(() => {
+          who === VET
+            ? dispatch(searchVets(request))
+            : dispatch(searchOwners(request))
+        }, 700)
+      )
+    }
+  }, [request])
 
-    if (timeout) clearTimeout(timeout)
-    setTimeoutState(setTimeout(() => dispatch(searchVets(request)), 700))
+  const getPages = () => {
+    return who === VET
+      ? ceil(
+          searchState.vetsResponse.totalItems /
+            searchState.vetsResponse.pageSize
+        )
+      : ceil(
+          searchState.ownersResponse.totalItems /
+            searchState.ownersResponse.pageSize
+        )
+  }
+
+  const onInputChange = (e: any) => {
+    setRequest({ ...request, query: e.target.value })
   }
 
   const renderResults = () => (
     <>
-      {renderVets(searchState.vetsResponse)}
-      {renderOwners(searchState.ownersResponse)}
+      {who === VET ? (
+        <h2 className={styles.resultTitle}>Weterynarze</h2>
+      ) : (
+        <h2 className={styles.resultTitle}>Użytkownicy</h2>
+      )}
+      <br />
+      <div className={styles.resultContainer}>
+        {who === VET
+          ? renderVets(searchState.vetsResponse)
+          : renderOwners(searchState.ownersResponse)}
+      </div>
     </>
   )
 
   return (
     <div className={styles.container}>
       <div className={styles.inputContainer}>
+        <RadioGroup
+          row
+          defaultValue={VET}
+          onChange={e => setWho(e.target.value)}
+        >
+          <FormControlLabel
+            value={VET}
+            control={<Radio color="primary" />}
+            label="Weterynarze"
+          />
+          <FormControlLabel
+            value={OWNER}
+            control={<Radio color="primary" />}
+            label="Opiekunowie"
+          />
+        </RadioGroup>
         <TextField
           className={styles.input}
           type="search"
           label="Wyszukaj"
           placeholder="Wyszukaj"
           inputMode="search"
           onChange={onInputChange}
+          onKeyPress={e => {
+            if (e.which === 13) onInputChange(e)
```

Comment:
czy szukajka działa dopiero po wciśnięciu entera czy moze robimy live search? 

Chyba że jest to sorki xD

Reply:
jest live search ale dodatkowo enter jak chcesz przeładować ale w sumie to już nie jest potrzebne, bo zrobiłem szacher macher i działa wszystko kozacko